okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -581,6 +604,7 @@ sslSocketFactory, hostnameVerifier, sock, getOverridenHost(), getOverridenPort()
           if (sslSession != null) {
             securityInfo = new InternalChannelz.Security(new InternalChannelz.Tls(sslSession));
           }
+          lock.notify();
```

Comment:
This is only happening in the successful case. We'd want to release the other thread in error cases as well.